lib/managers/folders.js

```diff
@@ -301,6 +301,35 @@ Folders.prototype.deleteMetadata = function(folderID, scope, template, callback)
 	this.client.del(apiPath, null, this.client.defaultResponseHandler(callback));
 };
 
+/**
+ * Restores an item that has been moved to the trash. Default behavior is to restore the item
+ * to the folder it was in before it was moved to the trash. If that parent folder no longer
+ * exists or if there is now an item with the same name in that parent folder, the new parent
+ * older and/or new name will need to be included in the request.
+ *
+ * API Endpoint: '/folders/:folderID'
+ * Method: POST
+ *
+ * @param {string} folderID - The ID of the folder to restore
+ * @param {string} name - The new name for the folder
```

Comment:
This parameter is optional!  We denote that in JSDoc by putting square brackets around the param name or by putting a question mark before the type (to indicate nullability).  In this case, since it's possible for name to be left blank but a new parent folder ID to be provided, I think it's best to make the param nullable and optional, e.g. `@param {?string} [name]`